drivers/sensor/bh1749/bh1749.c

```diff
@@ -0,0 +1,332 @@
+			LOG_ERR("Could not enable sensor interrupt.");
+			return -EIO;
+		}
+
+		/* Verify that INT_ENABLE is set */
+		u8_t interrupt_enabled;
+		if (i2c_reg_read_byte(data->i2c, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				      BH1749_INTERRUPT, &interrupt_enabled)) {
+			LOG_ERR("Could not enable sensor interrupt.");
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static int bh1749_channel_get(struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct bh1749_data *data = dev->driver_data;
+
+	switch (chan) {
+	case SENSOR_CHAN_RED:
+		val->val1 = sys_le16_to_cpu(
+			data->sample_rgb_ir[BH1749_RGB_BYTE_POS_RED]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_GREEN:
+		val->val1 = sys_le16_to_cpu(
+			data->sample_rgb_ir[BH1749_RGB_BYTE_POS_GREEN]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_BLUE:
+		val->val1 = sys_le16_to_cpu(
+			data->sample_rgb_ir[BH1749_RGB_BYTE_POS_BLUE]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_IR:
+		val->val1 = sys_le16_to_cpu(
+			data->sample_rgb_ir[BH1749_RGB_BYTE_POS_IR]);
+		val->val2 = 0;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int bh1749_check(struct device *dev)
+{
+	int err;
+	struct bh1749_data *data = dev->driver_data;
+
+	u8_t manufacturer_id;
+
+	err = i2c_reg_read_byte(data->i2c, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				BH1749_MANUFACTURER_ID, &manufacturer_id);
+	if (err) {
+		LOG_ERR("Failed when reading manufacturer ID");
+		return -EIO;
+	}
+
+	LOG_DBG("Manufacturer ID: 0x%02x", manufacturer_id);
+
+	if (manufacturer_id != BH1749_MANUFACTURER_ID_DEFAULT) {
+		LOG_ERR("Invalid manufacturer ID: 0x%02x", manufacturer_id);
+		return -EIO;
+	}
+
+	u8_t part_id;
+
+	err = i2c_reg_read_byte(data->i2c, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				BH1749_SYSTEM_CONTROL, &part_id);
+	if (err) {
+		LOG_ERR("Failed when reading part ID");
+		return -EIO;
+	}
+
+	if ((part_id & BH1749_SYSTEM_CONTROL_PART_ID_Msk) !=
+	    BH1749_SYSTEM_CONTROL_PART_ID) {
+		LOG_ERR("Invalid part ID: 0x%02x", part_id);
+		return -EIO;
+	}
+
+	LOG_DBG("Part ID: 0x%02x", part_id);
+
+	return 0;
+}
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int bh1749_power_control(struct device *dev, u32_t ctrl_command,
+				void *context)
+{
+	struct bh1749_data *data = dev->driver_data;
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		u32_t device_pm_state = *(u32_t *)context;
+
+		if (device_pm_state == DEVICE_PM_ACTIVE_STATE) {
+			if (i2c_reg_update_byte(data->i2c,
+						DT_ROHM_BH1749_0_BASE_ADDRESS,
+						BH1749_MODE_CONTROL2,
+						BH1749_MODE_CONTROL2_RGB_EN_Msk,
+						BH1749_MODE_CONTROL2_RGB_EN_ENABLE)) {
+				return -EIO;
+			}
+
+			return 0;
+		} else if (device_pm_state == DEVICE_PM_LOW_POWER_STATE) {
+			if (i2c_reg_update_byte(data->i2c,
+						DT_ROHM_BH1749_0_BASE_ADDRESS,
+						BH1749_MODE_CONTROL2,
+						BH1749_MODE_CONTROL2_RGB_EN_Msk,
+						BH1749_MODE_CONTROL2_RGB_EN_DISABLE)) {
+				return -EIO;
+			}
+
+			return 0;
+		}
```

Comment:
The sensor doesn't support any other states.

Reply:
Perhaps we could return `-ENOTSUP` for those not supported

Reply:
Since https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/master/soc/arm/nordic_nrf/Kconfig.defconfig#L32 sets default SYS_POWER_MANAGEMENT=y, which creates an error at https://github.com/zephyrproject-rtos/zephyr/blob/b58aa20e135a87dec10cb6814f323451ffe7565a/subsys/power/device.c#L25 when using nRF91x, I think that this is out of scope for this PR.
Disabling SYS_PM got me in to a world of bootup hardfaults, so I won't fight with that anymore for now.